File: snow_gushi/question1/KiwiJuiceEasy.py
class KiwiJuiceEasy(object):
    def thePouring(self, capacities, bottles, fromId, toId):
        capacities = list(capacities)
        bottles = list(bottles)
        fromId = list(fromId)
        toId = list(toId)
        for index, each_from_id in enumerate(fromId):
            f = fromId[index]
            t = toId[index]
            space = capacities[t] - bottles[t]

            if ( space >= bottles[f]):
                vol = bottles[f]
                print(vol)
                print(bottles[t])
                bottles[t] += vol
                bottles[f] = 0
            else:
                vol = space
                bottles[t] += vol
                bottles[f] -= vol

        return bottles

File: chooblarin/question01/KiwiJuiceEasy.java
public class KiwiJuiceEasy {
    public int[] thePouring(int[] capacities, int[] bottles, int[] fromId, int[] toId) {
        for (int i = 0; i < fromId.length; i++) {
            int from = fromId[i];
            int to = toId[i];
            if (capacities[to] < bottles[to] + bottles[from]) {
                int poured = Math.min(capacities[to] - bottles[to], bottles[from]);
                bottles[from] -= poured;
                bottles[to] += poured;
            } else {
                bottles[to] += bottles[from];
                bottles[from] = 0;
            }
        }
        return bottles;
    }
}
